Use 'istioctl dashboard' instead of port-forward

esnible opened the pull request:
Mitigates https://github.com/istio/istio.io/issues/4954

Note that the original request was for the dashboard docs, which this doesn't fix.

@geeknoid Currently `istioctl dashboard` picks a random local port.  This avoids conflicts, but it makes it impossible to directly link to the localhost URL in the documentation.  To finish this work item I am considering a change so that `dashboard` attempts to grab the well-known port number, and only picks a random one if the port is already busy.  That would minimize the doc changes.  What do you think?

istio-testing commented:
@esnible: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io | 32b866a80fcb28e723c1ff4c21a7040a5c90ec68 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/5296/lint_istio.io/916) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

howardjohn commented on content/en/docs/ops/common-problems/observability-issues/index.md:
```diff
@@ -174,10 +174,10 @@ in processing the most recent configuration as supplied.
     {{< /text >}}
 
-1. Visit `http://localhost:9090/targets`
+1. In the Prometheus browser window, select Status then Targets
```
Below we specify to go to :9090 again

howardjohn commented on content/en/docs/ops/common-problems/observability-issues/index.md:
```diff
@@ -174,10 +174,10 @@ in processing the most recent configuration as supplied.
 
     {{< text bash >}}
-    $ kubectl -n istio-system port-forward $(kubectl -n istio-system get pod -l app=prometheus -o jsonpath='{.items[0].metadata.name}') 9090:9090 &
+    $ istioctl dashboard prometheus &
```
```suggestion
    $ istioctl dashboard prometheus
```

For consistency - no strong opinion on this though

istio-testing commented:
@esnible: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io | 8ff88792ee5b833a59cd96143332010916a01de2 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/5296/lint_istio.io/920) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@esnible: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io | 88e298b034b19f404c848822367977129523c691 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/5296/lint_istio.io/930) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

geeknoid commented on content/en/docs/tasks/observability/distributed-tracing/jaeger/index.md:
```diff
@@ -65,10 +63,10 @@ Open your browser to [http://localhost:15032](http://localhost:15032).
 ## Cleanup
 
-1.  Remove any `kubectl port-forward` processes that may still be running:
+1.  Remove any `istioctl dashboard` processes that may still be running:
```
```suggestion
1.  Remove any `istioctl` processes that may still be running:
```

geeknoid commented on content/en/docs/tasks/observability/distributed-tracing/zipkin/index.md:
```diff
@@ -65,10 +63,10 @@ Open your browser to [http://localhost:15032](http://localhost:15032).
 ## Cleanup
 
-1.  Remove any `kubectl port-forward` processes that may still be running:
+1.  Remove any `istioctl dashboard` processes that may still be running:
```
```suggestion
1.  Remove any `istioctl` processes that may still be running:
```

geeknoid commented on content/en/docs/tasks/observability/metrics/querying-metrics/index.md:
```diff
@@ -108,10 +108,10 @@ docs](https://prometheus.io/docs/querying/basics/).
 ## Cleanup
 
--   Remove any `kubectl port-forward` processes that may still be running:
+-   Remove any `istioctl dashboard` processes that may still be running:
```
```suggestion
-   Remove any `istioctl` processes that may still be running:
```

geeknoid commented:
@esnible Do you want to land this PR for 1.4?

esnible commented:
I have addressed the comments and fixed the conflict.

Remaining is improving deep links to Grafana.  I propose waiting for https://github.com/istio/istio/pull/18552 to merge and then doing a follow-up.
